### PR TITLE
More docs for git references

### PR DESCRIPTION
**Pull request by kshyatt**

### CODE REVIEW

File: base/libgit2/reference.jl [hunk 1]
```diff
@@ -30,24 +30,49 @@ function isorphan(repo::GitRepo)
...
     @check ccall((:git_repository_head, :libgit2), Cint,
                   (Ptr{Ptr{Void}}, Ptr{Void}), head_ptr_ptr, repo.ptr)
     return GitReference(repo, head_ptr_ptr[])
 end
 
+"""
+    LibGit2.shortname(ref::GitReference)
+
+Returns a shortened version of the name of `ref` that's
+"human-readable".
```
Comment (tkelman) [inline base/libgit2/reference.jl]: in what way? is there an xref to the libgit2 or git-git docs that explains what they mean by this?

File: base/libgit2/reference.jl [hunk 2]
```diff
@@ -30,24 +30,49 @@ function isorphan(repo::GitRepo)
...
+"""
 function reftype(ref::GitReference)
     return ccall((:git_reference_type, :libgit2), Cint, (Ptr{Void},), ref.ptr)
 end
 
+"""
+    LibGit2.fullname(ref::GitReference)
+
+Return the name of the reference pointed to by the
+symbolic reference `ref`.
```
Comment (tkelman) [inline base/libgit2/reference.jl]: Add something to the effect of "If `ref` is not a symbolic reference [or whatever else could cause the ccall to return a null pointer?] then returns the empty string."

File: base/libgit2/reference.jl (hunk 1 above)
Comment (kshyatt) [inline base/libgit2/reference.jl]: This is about all the `libgit2` docs give us...

File: base/libgit2/reference.jl (hunk 2 above)
Comment (kshyatt) [inline base/libgit2/reference.jl]: will do :)

File: base/libgit2/reference.jl [hunk 3]
```diff
@@ -141,6 +176,15 @@ function ref_list(repo::GitRepo)
     res
 end
 
+"""
+    LibGit2.create_branch(repo::GitRepo, bname::AbstractString, commit_obj::GitCommit; force::Bool=false)
+
+Create a new branch in the repository `repo` with name `bname`, which
+points to commit `commit_obj` (which has to be part of `repo`). If
+`force` is `true`, overwrite an existing branch with `bname` if it
```
Comment (tkelman) [inline base/libgit2/reference.jl]: maybe "overwrite an existing branch named `bname` if it exists."

Comment (kshyatt) [inline base/libgit2/reference.jl]: Got this

File: base/libgit2/reference.jl (hunk 2 above)
Comment (kshyatt) [inline base/libgit2/reference.jl]: Got this too

File: base/libgit2/reference.jl (hunk 1 above)
Comment (kshyatt) [inline base/libgit2/reference.jl]: Should I make an example (probably not a doctest) just to show this?

Comment (tkelman) [inline base/libgit2/reference.jl]: good idea

**Comment by kshyatt** [comment]
@tkelman ready to go?

File: base/libgit2/reference.jl [hunk 4]
```diff
@@ -132,6 +175,11 @@ function peel{T<:GitObject}(::Type{T}, ref::GitReference)
 end
 peel(ref::GitReference) = peel(GitObject, ref)
 
+"""
+    LibGit2.ref_list(repo::GitRepo)
```
Comment (tkelman) [inline base/libgit2/reference.jl]: `-> Vector{String}`

**Pull request merged**